client/src/components/deleteBlog/index.jsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/prop-types */
+import React, {useState} from 'react'
+import { Redirect } from 'react-router-dom';
+import { Button, Modal } from 'semantic-ui-react'
+import { deleteBlog } from '../../apis/blogs';
+
+const DeleteBlog = ({triggerButton, blogId}) => {
+    const [deleting, updateDeleting] = useState(false);
+    const [deleted, updateDeleted] = useState(false);
+    const [error, updateError] = useState(null);
+
+    const confirmDelete = () => {
+        updateDeleting(true);
+        deleteBlog(blogId).then(()=>{
+            updateDeleted(true) 
+        }).catch((e) => {
+            updateError(e);
+        })
+    }
+    if(deleted) {
+        return (
+            <Redirect 
+            to="/"
+            />
+        )
+    }
+  return (<Modal
+  size="mini"
+   trigger={triggerButton || <Button color="red" style={{marginLeft:'10px'}} >Delete Blog</Button>}
+   >
+    <Modal.Header>Are you sure you want to delete the blog?</Modal.Header>
+    <Modal.Content>
+      <Modal.Description>
+        <p>You won&apos;t be able to undo this action.</p>
+        <Button color="red" 
+        disabled={deleting}
+        content="Confirm Delete"
+        icon="trash"
+         onClick={confirmDelete} 
+        loading={deleting}
+         />
+         {
```

Comment:
Add button as Modal.Action trust me it will look good. 2 line change.

Reply:
Okay wait

Reply:
https://react.semantic-ui.com/modules/modal/#types-controlled
check this